lib/tools/src/tprof.erl

```diff
@@ -868,36 +856,42 @@ profile(Module, Function, Args, Options) when is_atom(Module), is_atom(Function)
 init(Config) ->
     Type = maps:get(type, Config, call_count),
     false = erlang:process_flag(trap_exit, true), %% need this for reliable terminate/2 call
-    {ok, #tprof_state{type=Type}}.
+    Session = trace:session_create(maps:get(session_name, Config, ?MODULE), self(), []),
```

Comment:
Dialyzer probably does not see that, but `session_name` appears to be what `registered_name` used to stand for before this change. I think this behaviour should be documented.

Now that we have multiple trace sessions support, there is not much sense in the trick original `hprof` did. We can have many `tprof` sessions running, identified by PID.

Reply:
> Dialyzer probably does not see that, but session_name appears to be what registered_name used to stand for before this change. I think this behaviour should be documented.

The session name does not have to be unique per node, so it is somewhat different from `registered_name`. The tprof server is registered as `tprof` when not using the oneshot API, but if we do use the oneshot we can have as many sessions as we want in parallel.

Reply:
I mean, even if it's different from the `registered_name`, it seems worth documenting.